.github/workflows/ci-cd.yml

```diff
@@ -0,0 +1,65 @@
+name: CI/CD Workflow
+
+on:
+  push:
+    branches:
+      - 'main'
+      - 'feature/*'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: '20'
+
+      - name: Install dependencies using pnpm
+        run: |
+          corepack enable
+          pnpm install
+
+      - name: Run tests
+        run: pnpm test
+
+  deploy:
+    if: github.ref == 'refs/heads/main'
+    needs: build
```

Comment:
Is the dependency to `build`-step required? The same steps are executed anyways again